src/features/common/AuthFireRoute.js

```diff
@@ -1,22 +1,58 @@
 import { Redirect, Route } from "react-router-dom";
-
 import PropTypes from "prop-types";
 import React from "react";
 import { useAuth } from "../../contexts/AuthContext";
 
-export default function AuthFirebaseRoute({ component: Component, ...rest }) {
+//AuthFirebaseRouteはログインの有無によって遷移ページを変える関数
+//ログインしていない場合はリダイレクトされる
+const AuthRouteRedirect = ({ component: Component, path: path }) => {
   const { currentUser } = useAuth();
   if (!currentUser) {
     return <Redirect to="/login" />;
   } else {
     return (
-      <Route {...rest}>
+      <Route path={path}>
+        <Component />
+      </Route>
+    );
+  }
+};
+
+//AuthRouteGuestPageはログインの有無によって遷移ページを変える関数
+//ログインしていない場合はゲストページへ遷移する
+export const AuthRouteGuestPage = ({
+  component: Component,
+  componentGuest: ComponentGuest,
+  path: path,
+  pathGuest: pathGuest,
+}) => {
+  const { currentUser } = useAuth();
+
+  if (!currentUser) {
+    return (
+      <Route path={pathGuest}>
+        <ComponentGuest />
+      </Route>
+    );
+  } else {
+    return (
+      <Route path={path}>
```

Comment:
```suggestion
      <Route path={path} component={Component} />
```

で良いかも．App.jsもそのうち修正したほうがいいですね．

https://reactrouter.com/web/api/Route/component